src/Components/addroom/AddRoom.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import classes from './AddRoom.module.css';
+import { createRoom } from '../../redux/store';
+
+export default function AddRoom() {
+  const [state, setState] = useState({
+    picture: '',
+    name: '',
+    number_of_beds: '',
+    price: '',
+    description: '',
+    user_id: '',
+    room_type_id: '',
+  });
+
+  const dispatch = useDispatch();
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setState((prevProps) => ({
+      ...prevProps,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const room = {
+      name: event.target.elements.name.value,
+      picture: event.target.elements.picture.value,
+      number_of_beds: event.target.elements.number_of_beds.value,
+      price: event.target.elements.price.value,
+      description: event.target.elements.description.value,
+      user_id: event.target.elements.user_id.value,
+      room_type_id: event.target.elements.room_type_id.value,
+    };
+    dispatch(createRoom(room));
+  };
+
+  return (
+    <div className={classes.add_book_dev}>
+      <form onSubmit={handleSubmit}>
+        <div className={classes.form}>
+          <label htmlFor="user_id">User Id</label>
+          <input
+            type="text"
+            name="user_id"
+            value={state.user_id}
+            onChange={handleInputChange}
+          />
+        </div>
+        <div className={classes.form}>
+          <label htmlFor="room_type_id">Room Type Id</label>
+          <input
+            type="text"
+            name="room_type_id"
+            value={state.room_type_id}
+            onChange={handleInputChange}
+          />
+        </div>
```

Comment:
- Please, make sure that the `room_type_id` is not an input field but that you can use a `select option` instead so that the user can select the `room_type` by name and not by inserting an unknown number. (i.e) Imagine a scenario where that hotel has hundred of `rooms type`, will you be able to recognize the `id` of each?